### PR TITLE
chore(docs): Add constructs documentation to website

### PR DESCRIPTION
**Pull request by laurapacilio**
This PR:
- Adds constructs documentation to the concepts section of the Terraform website. 
- Adds the constructs page to the relevant place in the sidebar
- Adds a linked reference on the architecture.html, providers-and-resources.html, and stacks.html pages

### CODE REVIEW

**Review by ansgarm — APPROVED** [review]

**Comment by laurapacilio** [comment]
@ansgarm once you and the team feel comfortable with this, would you mind merging? I just don't have permissions :-)

**Comment by ansgarm** [comment]
@laurapacilio I wanted to give @schersh a chance to read first, but we already talked about it and she'll merge it then :)

File: website/docs/cdktf/concepts/constructs.html.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+## Modules vs. Constructs
+
+Both Terraform [modules](/docs/cdktf/concepts/modules.html) and constructs allow you to reuse configurations and customize those configurations via exposed properties. However, constructs let you compose more complex objects while maintaining type safety because they can use the built-in functionality from your preferred programming language.
+
+### Interoperability
+
+You can use the [`cdktf convert`](/docs/cdktf/cli-reference/commands.html#convert) command to translate existing projects written in HashiCorp Configuration Language (HCL) into CDKTF-compatible projects. You could then use the output as a starting point to create custom constructs.
+
+You cannot create a Terraform module directly from an existing CDKTF construct, but you can use the synthesized output of a CDKTF project as a Terraform Module. Refer to the [HCL interoperability](website/docs/cdktf/concepts/hcl-interoperability.html.md) page for more details.
```
Comment (schersh) [inline website/docs/cdktf/concepts/constructs.html.md:37]:
```suggestion
You cannot create a Terraform module directly from an existing CDKTF construct, but you can use the synthesized output of a CDKTF project as a Terraform Module. Refer to the [HCL interoperability](/docs/cdktf/concepts/hcl-interoperability.html) page for more details.
```

File: website/docs/cdktf/concepts/constructs.html.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    // the same scope. Therefore, their names must be unique.
+    new PublicS3Bucket(this, "first-bucket");
+    new PublicS3Bucket(this, "second-bucket");
+  }
+}
+```
+
+### Aspects
+
+`Aspects` allow you to implement a visitor pattern, dynamically add or remove constructs, and automatically change attributes of existing constructs when you synthesize your CDTKF application. For example, you could use an aspect to help tag resources based on dynamic conditions. Refer to the [aspects documentation](docs/cdktf/concepts/aspects.html) for more details.
```
Comment (schersh) [inline website/docs/cdktf/concepts/constructs.html.md:125]:
```suggestion
`Aspects` allow you to implement a visitor pattern, dynamically add or remove constructs, and automatically change attributes of existing constructs when you synthesize your CDTKF application. For example, you could use an aspect to help tag resources based on dynamic conditions. Refer to the [aspects documentation](/docs/cdktf/concepts/aspects.html) for more details.
```

File: website/docs/cdktf/concepts/constructs.html.md (hunk 1 above)
Comment (laurapacilio) [reply website/docs/cdktf/concepts/constructs.html.md:37]: thank you for catching this!

File: website/docs/cdktf/concepts/constructs.html.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+### Available Constructs
+
+You can search the Amazon Web Services (AWS) [Construct Hub](https://constructs.dev/search?q=&cdk=cdktf&sort=newest&offset=0) for existing CDKTF-compatible constructs. We are also building the [AWS Adapter](/docs/cdktf/create-and-deploy/aws-adapter.html), which lets you use AWS Constructs in your CDKTF projects.
```
Comment (schersh) [inline website/docs/cdktf/concepts/constructs.html.md:129]:
```suggestion
You can search the Amazon Web Services (AWS) [Construct Hub](https://constructs.dev/search?q=&cdk=cdktf&offset=0) for existing CDKTF-compatible constructs. We are also building the [AWS Adapter](/docs/cdktf/create-and-deploy/aws-adapter.html), which lets you use AWS Constructs in your CDKTF projects.
```

**Review by schersh — APPROVED** [review]
Thanks, approving with a few link fixes.

**Pull request merged**